### PR TITLE
fix: improve the light-theme 

### LINKED ISSUE #2247: [Feature] improvements in light and dark theme
### Description

@rupali-codes
I'm creating this issue, where we can further improve the light theme.

Feel free to update this issue, or the points.

---

1. A bit of a border and small shadow box.

| White (to improve) | Dark (inspiration) |
|:---------------------:|:--------------------:|
| image | image |

2. Hover color changes on selected links. 

image

3. A bit of a border and small shadow box.

| White (to improve) | Dark (inspiration) |
|:---------------------:|:--------------------:|
| image | image |

4. Improve breadcrumbs.

| White (to improve) | Dark (inspiration) |
|:---------------------:|:--------------------:|
| image | image |

5. A border on the searchbar

| White (to improve) | Dark (inspiration) |
|:---------------------:|:--------------------:|
| image | image |

6. A border and shadow box.

image

---

@aftabrehan @CBID2 
Feel free to suggest whatever you think!

### Screenshots

> Next time :)

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [x] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by CBID2**
## Fixes Issue
Closes #2247 

## Changes proposed
This PR changes the website's light theme, making it easier for people to navigate the site. Here's what's been done so far:

- [x] Made the border of the searchbar's input darker and thicker (see the new version of the first screenshot for reference.)
- [x] Made the border of the breadcrumbs darker and thicker (see the second screenshot for reference.)
- [x] Add shadowbox and border to card-list and footer(see third screenshot for reference)

## Screenshots
1. Search input having a dark input
screenshot of search input with a darker border
**Update** here's a new version:
new version of light theme

2. Breadcrumbs having a darker and thicker border
made the breadcrumb's light theme darker and thicker
**Update** here's an new version of the breadcrumb
new version of breadcrumbs

3. Footer and card list
footer and card

## Note to reviewers
This is a work-in-progress

### CODE REVIEW

**Comment by Anmol-Baranwal** [comment]
@CBID2 

I know the draft PR isn't ready, but I would suggest these changes.

1. Even if you want a border in dark theme, use the light border that is used in the cards as shown.

image

2. and the same for light background. Use a little light border if possible (use opacity). 
It shouldn't be the reason of distraction.

image

**Comment by CBID2** [comment]
> @CBID2 
> 
> 
> 
> I know the draft PR isn't ready, but I would suggest these changes.
> 
> 
> 
> 1. Even if you want a border in dark theme, use the light border that is used in the cards as shown.
> 
> 
> 
> image
> 
> 
> 
> 2. and the same for light background. Use a little light border if possible (use opacity). 
> 
> It shouldn't be the reason of distraction.
> 
> 
> 
> image
> 
> 

@Anmol-Baranwal, so...something like this: https://github.com/rupali-codes/LinksHub/blob/main/components/Cards/Card.tsx

**Comment by Anmol-Baranwal** [comment]
> @Anmol-Baranwal, so...something like this: https://github.com/rupali-codes/LinksHub/blob/main/components/Cards/Card.tsx `border-light-white`

For light, I'm not sure if there is any. But for dark, using the one on the cards will avoid mismatch of theme.
You can check locally.

**Comment by sarava338** [comment]
In Dark theme, we have very dark color in background. But in Light theme, We did not. So I hope, the very light color in background and swap the with card's background color will do the trick.

Dark Theme
image

Light Theme
image

**Comment by CBID2** [comment]
> In Dark theme, we have very dark color in background. But in Light theme, We did not. So I hope, the very light color in background and swap the with card's background color will do the trick.
> 
> 
> 
> Dark Theme
> 
> image
> 
> 
> 
> Light Theme
> 
> image
> 
> 

So…you suggest that I use the dark theme’s background as a border for the light one @sarava338?

**Comment by Anmol-Baranwal** [comment]
> So…you suggest that I use the dark theme’s background as a border for the light one @sarava338?

He is saying that in dark theme -> The cards are light color compared to the background color.

But it's the opposite in the light theme, so if we make background complete white and cards as the background color (which is lighter one). Then this problem would be fixed.

@rupali-codes 
Don't do this without consulting rupali.

**Comment by sarava338** [comment]
> > So…you suggest that I use the dark theme’s background as a border for the light one @sarava338?
> 
> He is saying that in dark theme -> The cards are light color compared to the background color.
> 
> But it's the opposite in the light theme, so if we make background complete white and cards as the background color (which is lighter one). Then this problem would be fixed.
> 
> @rupali-codes 
> Don't do this without consulting rupali.

Yes. And in the Light theme, the little dark color has to be a little darker.

**Comment by CBID2** [comment]
> > So…you suggest that I use the dark theme’s background as a border for the light one @sarava338?
> 
> He is saying that in dark theme -> The cards are light color compared to the background color.
> 
> But it's the opposite in the light theme, so if we make background complete white and cards as the background color (which is lighter one). Then this problem would be fixed.
> 
> @rupali-codes Don't do this without consulting rupali.

Ooh just saw this @Anmol-Baranwal after adding the new changes (see the updated photos in the screenshot section of my PR's form). Hopefully, @rupali-codes won't mind the changes.

**Review by rupali-codes — COMMENTED** [review]
I don't see any changes except the border on searchbar

This PR
image

Main site
image

**Comment by CBID2** [comment]
> I don't see any changes except the border on searchbar
> 
> 
> 
> This PR
> 
> image
> 
> 
> 
> Main site
> 
> image
> 
> 
> 
> 
> 
> 
> 
> 

It's a work-in-progress @rupali-codes. Still trying to figure out how to do the other parts.

**Comment by rupali-codes** [comment]
@CBID2 alright, let us know if you need any help

**Comment by CBID2** [comment]
> @CBID2 alright, let us know if you need any help

Hey @rupali-codes and @Anmol-Baranwal, I’m struggling with adding the border to the card components. Can you help me please?

**Comment by rupali-codes** [comment]
> > @CBID2 alright, let us know if you need any help
> 
> Hey @rupali-codes and @Anmol-Baranwal, I’m struggling with adding the border to the card components. Can you help me please?

where you wanna add border?

**Comment by CBID2** [comment]
> > > @CBID2 alright, let us know if you need any help
> 
> > 
> 
> > Hey @rupali-codes and @Anmol-Baranwal, I’m struggling with adding the border to the card components. Can you help me please?
> 
> 
> 
> where you wanna add border?
> 
> 

The card components @rupali-codes

**Comment by rupali-codes** [comment]
> > > > @CBID2 alright, let us know if you need any help
> > 
> > 
> > > 
> > 
> > 
> > > Hey @rupali-codes and @Anmol-Baranwal, I’m struggling with adding the border to the card components. Can you help me please?
> > 
> > 
> > where you wanna add border?
> 
> The card components @rupali-codes

alright, it should be in `components/cards/card.tsx` file, then in the main `div` add border classes. 

What else are you struggling with?

**Comment by CBID2** [comment]
> > > > > @CBID2 alright, let us know if you need any help
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > Hey @rupali-codes and @Anmol-Baranwal, I’m struggling with adding the border to the card components. Can you help me please?
> > > 
> > > 
> > > where you wanna add border?
> > 
> > 
> > The card components @rupali-codes
> 
> alright, it should be in `components/cards/card.tsx` file, then in the main `div` add border classes.
> 
> What else are you struggling with?

Hi @rupali-codes. I added [the border to the main `div`](https://github.com/rupali-codes/LinksHub/blob/5acd6b910afed9f5bae2e1ff2582db0d6ab80390/components/Cards/Card.tsx#L31) like you suggested and nothing shows
 Did I do something wrong?

**Comment by rupali-codes** [comment]
> > > > > > @CBID2 alright, let us know if you need any help
> > > > 
> > > > 
> > > > > 
> > > > 
> > > > 
> > > > > Hey @rupali-codes and @Anmol-Baranwal, I’m struggling with adding the border to the card components. Can you help me please?
> > > > 
> > > > 
> > > > where you wanna add border?
> > > 
> > > 
> > > The card components @rupali-codes
> > 
> > 
> > alright, it should be in `components/cards/card.tsx` file, then in the main `div` add border classes.
> > What else are you struggling with?
> 
> Hi @rupali-codes. I added [the border to the main `div`](https://github.com/rupali-codes/LinksHub/blob/5acd6b910afed9f5bae2e1ff2582db0d6ab80390/components/Cards/Card.tsx#L31) like you suggested and nothing shows Did I do something wrong?

You've misplaced the double quotes 

image


It should be ` <div className="card-body border-light-silver border-2 border-opacity-75">`

**Comment by Anmol-Baranwal** [comment]
@CBID2 
This is still in draft. Are the changes done.

**Comment by CBID2** [comment]
> @CBID2 
> 
> This is still in draft. Are the changes done.

No I have to do the footer @Anmol-Baranwal

**Comment by rupali-codes** [comment]
> > @CBID2
> > This is still in draft. Are the changes done.
> 
> No I have to do the footer @Anmol-Baranwal

you need any help Chrissy?

**Comment by CBID2** [comment]
> > > @CBID2
> 
> > > This is still in draft. Are the changes done.
> 
> > 
> 
> > No I have to do the footer @Anmol-Baranwal
> 
> 
> 
> you need any help Chrissy?

Not yet @rupali-codes

**Comment by CBID2** [comment]
Hey @rupali-codes. A huge merge conflict happened. Did `git rebase`, a force push happened, so I had to do a redo. Trying to add the border to the breadcrumbs file but this happened:
## border 
border
## No border
no border

**Comment by rupali-codes** [comment]
> Hey @rupali-codes. A huge merge conflict happened. Did `git rebase`, a force push happened, so I had to do a redo. Trying to add the border to the breadcrumbs file but this happened:
> 
> ## border
> border
> 
> ## No border
> no border

i think it looks fine with no border

**Comment by rupali-codes** [comment]
@CBID2 why is it showing like this 😅

image

**Comment by CBID2** [comment]
> @CBID2 why is it showing like this 😅
> 
> 
> 
> image
> 
> 

Made an error lol

**Comment by CBID2** [comment]
> @CBID2, let me know if you need any help with this?

Thanks @aftabrehan. I can't remove the larger border in the card list: https://github.com/rupali-codes/LinksHub/pull/2256#issuecomment-1961351849

**Comment by aftabrehan** [comment]
> > @CBID2, let me know if you need any help with this?
> 
> Thanks @aftabrehan. I can't remove the larger border in the card list: [#2256 (comment)](https://github.com/rupali-codes/LinksHub/pull/2256#issuecomment-1961351849)

Let me have a look.

File: components/Cards/CardsList.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ const CardsList: FC<{ cards: IData[] }> = ({ cards }) => {
 
   return (
     <>
-      <ul className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-4 justify-items-stretch ">
+      <ul className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-4 justify-items-stretch border-dark-silver border-2 border-opacity-75 shadow-lg shadow-silver">
```
Comment (aftabrehan) [inline components/Cards/CardsList.tsx:22]: @CBID2, here is the fix.

```suggestion
      <ul className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-4 justify-items-stretch">
```

**Comment by aftabrehan** [comment]
@CBID2, please make sure to pull the latest changes before making any adjustments.

**Comment by aftabrehan** [comment]
@rupali-codes, can you help deploy the changes?

**Comment by CBID2** [comment]
Hey @rupali-codes. I finished everything. Can you authorize the deployment so you, @Anmol-Baranwal, and @aftabrehan can review the latest changes?

File: components/Cards/Card.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export const Card: FC<CardProps> = ({ data, onClick }) => {
 
   return (
     <article className="z-10 h-full w-full rounded-3xl dark:bg-slate-800 dark:text-text-primary dark:shadow-sm bg-light-white">
-      <div className="card-body">
+      <div className="card-body border-light-silver border-2 border-opacity-75">
```
Comment (aftabrehan) [inline components/Cards/Card.tsx:31]: @CBID2, I think we should add the border on the card parent wrapper.

```suggestion
    <article className="z-10 h-full w-full rounded-3xl dark:bg-slate-800 dark:text-text-primary dark:shadow-sm bg-light-white border-light-silver border-2 border-opacity-75">
      <div className="card-body">
````

Screenshot 2024-03-03 at 11 44 10 AM

File: components/Searchbar/Searchbar.tsx [hunk 3]
```diff
@@ -114,7 +114,7 @@ export const Searchbar: React.FC<SearchbarProps> = ({
       role="search"
     >
       <div className="relative">
-        <div className="relative w-full h-12 flex items-center justify-between">
+        <div className="relative w-full h-12 flex items-center justify-between border-light-silver border-2 border-opacity-75">
```
Comment (aftabrehan) [inline components/Searchbar/Searchbar.tsx:117]: @CBID2, Please, remove the border from here and add on the input element.

**Review by aftabrehan — CHANGES_REQUESTED** [review]

**Comment by CBID2** [comment]
Made the changes @aftabrehan

**Review by aftabrehan — APPROVED** [review]

**Review by rupali-codes — COMMENTED** [review]
This issue isn't getting fixed, been long ha,
when you try to fix it for light theme it gets weird for dark theme

image

image

@CBID2 please let me know if you need any help, wanna close this ASAP

**Comment by CBID2** [comment]
> This issue isn't getting fixed, been long ha,
> 
> when you try to fix it for light theme it gets weird for dark theme
> 
> 
> 
> image
> 
> 
> 
> image
> 
> 
> 
> @CBID2 please let me know if you need any help, wanna close this ASAP

Ok can you help @rupali-codes

**Comment by rupali-codes** [comment]
@CBID2 yes of course, where are you getting troubles?

**Comment by CBID2** [comment]
I'm not sure how to make it less weird looking in light mode @rupali-codes. From the photos you posted, it looks fine.

**Comment by CBID2** [comment]
What do you think @Anmol-Baranwal?

**Comment by Anmol-Baranwal** [comment]
> What do you think @Anmol-Baranwal?

I've no problem if others are okay with it.

**Comment by rupali-codes** [comment]
> @rupali-codes, I really think we should merge this.

Can you remove the border for dark theme and keep it only for light theme?

**Comment by CBID2** [comment]
> > @rupali-codes, I really think we should merge this.
> 
> 
> 
> Can you remove the border for dark theme and keep it only for light theme?

Ok

**Comment by CBID2** [comment]
Hey @rupali-codes. Can you do another deployment? I made the changes to light theme but I can't see them locally.

**Comment by CBID2** [comment]
@rupali-codes, can you deploy this please?

**Comment by Anmol-Baranwal** [comment]
@CBID2 
Should I create a new PR for this? It's urgent because the current live light theme (live) is far from good.

**Comment by CBID2** [comment]
> @CBID2 
> 
> Should I create a new PR for this? It's urgent because the current live light theme (live) is far from good.

Ok @Anmol-Baranwal

**Pull request closed without merging**